samplelistener/listener/dispatcher_service.bal

```diff
@@ -2,46 +2,43 @@ import ballerina/http;
 import ballerina/jballerina.java;
 
 service class DispatcherService {
-   private SlackAppMentionHandlingService|SlackAppCreatedHandlingService serviceRef;
-   //private string verificationToken;
+     private map<GenericService> services = {};
 
-   isolated function init(SlackAppMentionHandlingService|SlackAppCreatedHandlingService serviceRef) returns error? { //Receive verification token
-        self.serviceRef = serviceRef;
-        //self.verificationToken = "9asdbas9009123nas1e2";
+   isolated function addServiceRef(string serviceType, GenericService genericService) returns error? {
+        if (self.services.hasKey(serviceType)) {
+             return error("Same service type has been defined more than one time");
```

Comment:
```suggestion
             return error(string `service of type ${serviceType} has already been attached");
```

If there's more than one service type for which services have been attached more than once, with this approach, the user will get each error one at a time. Other alternative would be to keep track of the counts separately and provide this error at the beginning of the listening phase (i.e., when start is called) listing all the types with more than one service. We can discuss and do this separately though.

Reply:
Thanks for the feedback. Did the suggested change to the error. Noted the other point and let's discuss that as a separate thing. Thanks